src/Factories/TestCaseMethodFactory.php

```diff
@@ -29,7 +30,7 @@ final class TestCaseMethodFactory
     /**
      * The Test Case Dataset, if any.
      *
-     * @var array<Closure|iterable<int|string, mixed>|string>
+     * @var array<TestCaseDataset>
```

Comment:
Is it a `TestCaseDataSet` or a `TestDataSet`? I would say that given it only pertains to a single test, it should be the latter?

Reply:
Yeah, you're right, I think we should rename the class

Reply:
Refactored to `TestDataset`